lib/Elastica/Reindex.php

```diff
@@ -138,4 +162,62 @@ private function _setSourceType(array $sourceBody)
 
         return $sourceBody;
     }
+
+    private function _getEndpointParams(array $params)
+    {
+        return array_intersect_key($params, [
+            self::REFRESH => null,
+            self::WAIT_FOR_COMPLETION => null,
+            self::WAIT_FOR_ACTIVE_SHARDS => null,
+            self::TIMEOUT => null,
+            self::SCROLL => null,
+            self::REQUESTS_PER_SECOND => null,
+        ]);
+    }
+
+    public function setWaitForCompletion($value)
+    {
+        is_bool($value) && $value = $value ? 'true' : 'false';
+        $this->setParam(self::WAIT_FOR_COMPLETION, $value);
```

Comment:
normally we return `$this` so setters can be chained. Here it would be `return $this->setParam(...` Same below for the other methods.